components/director/internal/domain/tenant/repository.go

```diff
@@ -146,13 +150,48 @@ func (r *pgRepository) List(ctx context.Context) ([]*model.BusinessTenantMapping
 		return nil, errors.Wrap(err, "while listing tenants from DB")
 	}
 
-	items := make([]*model.BusinessTenantMapping, 0, len(entityCollection))
+	return r.multipleFromEntities(entityCollection), nil
+}
 
-	for _, entity := range entityCollection {
-		tmModel := r.conv.FromEntity(&entity)
-		items = append(items, tmModel)
+func (r *pgRepository) buildSearchCondition(fields []string, searchTerm string) string {
+	var stmtBuilder strings.Builder
+
+	lastIdx := len(fields) - 1
+
+	for idx, field := range fields {
+		if idx != lastIdx {
+			stmt := fmt.Sprintf(`%s ILIKE '%%%s%%' OR `, field, searchTerm)
+			stmtBuilder.WriteString(stmt)
+			continue
+		}
+
+		stmt := fmt.Sprintf(`%s ILIKE '%%%s%%'`, field, searchTerm)
+		stmtBuilder.WriteString(stmt)
+	}
```

Comment:
The arguments in the like operator must not be crafted by hand (via sprintf) in order to avoid SQL injection.
The right way to do this is to replace the `%s` with `$%d` and add the idx which will result in positional arguments.
Then you need an []interfaces representing the args and add all the searchTerms there.